utils/prometheus-node-exporter-ucode/files/metrics.uc

```diff
@@ -0,0 +1,187 @@
+{%
+'use strict';
+
+import * as fs from "fs";
+import { connect } from "ubus";
+
+// TODO reopen stdout as stderr
+function puts(...s) {
+	for (let i in s)
+		uhttpd.send(i);
+	uhttpd.send("\n");
```

Comment:
I tried that, but it doesn't work.

I guess because the uhttpd ucode plugin doesn't loop over the arguments?
https://git.openwrt.org/?p=project/uhttpd.git;a=blob;f=ucode.c;h=aa5d5b454fcd99ffbcb85f28249088512e6f6c0a;hb=HEAD#l123
vs
https://github.com/jow-/ucode/blob/master/lib.c#L277

May be related, what's the difference between send() and sendc()?

Reply:
The attached patch does exactly that, which makes this work. +10% perf, from 0.64 to 0.56.
[0001-ucode-respect-all-arguments-passed-to-send.patch.txt](https://github.com/openwrt/packages/files/10146010/0001-ucode-respect-all-arguments-passed-to-send.patch.txt)